extensions/wikia/Captcha/scripts/Captcha.js

```diff
@@ -0,0 +1,65 @@
+$(function () {
+	'use strict';
+
+	/**
+	 * JS for handling captchas We default to using reCaptcha, however if we encounter any problems, will
+	 * fall back to using Fancy Captcha instead.
+	 */
+	var Captcha = {
+		/**
+		 * WikiaMobile, Wikia One, and some automated tests do not use captcha
+		 */
+		useCaptcha: !window.wgUserSignupDisableCaptcha,
+
+		/**
+		 * Enable user signup form with ajax validation
+		 */
+		init: function () {
+			this.loadCaptcha();
+		},
+
+		loadCaptcha: function () {
+			if (this.useCaptcha) {
+				$.loadReCaptcha().fail(
+					this.handleCaptchaLoadError.bind(this)
+				);
+			}
+		},
+
+		/**
+		 * Captcha is required for signup, so if it fails to load (possibly b/c google is blocked in China)
+		 * inform the user with a modal. Note, this is different from when a user fails the captcha test itself.
+		 */
+		handleCaptchaLoadError: function () {
+			this.loadAndInsertFancyCaptcha();
+			this.trackFailedReCaptcha();
+		},
+
+		loadAndInsertFancyCaptcha: function () {
+			$.nirvana.sendRequest({
+				controller: 'CaptchaController',
+				method: 'getFancyCaptcha',
+				type: 'GET',
+				callback: function (data) {
+					this.insertFancyCaptcha(data);
+				}.bind(this)
+			});
+		},
+
+		insertFancyCaptcha: function (data) {
+			$('.g-recaptcha').replaceWith(data.form);
+		},
+
+		trackFailedReCaptcha: function () {
+			Wikia.Tracker.track({
+				action: Wikia.Tracker.ACTIONS.ERROR,
+				category: 'user-sign-up',
```

Comment:
This category is no longer accurate since this JS affects more than just user signup now. Maybe talk to Cristina to see if changing it would throw off her numbers.